### PR TITLE
revset: fix crash on "log --at-op 00000000 -r 'root()'"

### PR DESCRIPTION
**Pull request by yuja**
# Checklist

If applicable:
- [ ] I have updated `CHANGELOG.md`
- [ ] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (cli/src/config-schema.json)
- [x] I have added tests to cover my changes

### CODE REVIEW

File: lib/tests/test_revset.rs [hunk 1]
```diff
@@ -954,6 +963,14 @@ fn test_evaluate_expression_root_and_checkout() {
         vec![root_commit.id().clone()]
     );
 
+    // but not in the root operation. It might be okay to pretend that the root
+    // commit exists in the root operation, but queries like "root()" shouldn't
+    // panic in any case.
```
Comment (martinvonz) [inline lib/tests/test_revset.rs:968]: Yes, seems to me like it the root operation should have the root commit. I guess that would mean that the op store would have to learn about the root commit id, which I don't think it currently knows about.

**Review by martinvonz — APPROVED** [review]
Fun case

File: lib/tests/test_revset.rs (hunk 1 above)
Comment (martinvonz) [reply lib/tests/test_revset.rs:968]: But, to be clear, I'm not asking for any changes to this PR

Comment (yuja) [reply lib/tests/test_revset.rs:968]: Yes, it's a bit odd that `OpStore` knows about the root commit. If we had an abstraction layer on top of the backend `OpStore`, an empty view might be handled there.

**Pull request merged**